### DISCUSSION TITLE
"Random List" Option im Modul AnyStores Liste

### DISCUSSION THREAD

**Discussion by elcreativo** [Tastaturberuf/anyStores#90] [Feature Requests / Funktionswünsche] [answered]
Könnte man in das Listen Modul noch die Sortierung Random mit einbinden? Ich würde gerne auf der Startseite immer 4 Standorte zufällig anzeigen bzw. auch mehr in eine Art Slider.

**Comment by Tastaturberuf** [comment]
Das könnte man tun ja, im Moment kann man das übers Template lösen.

**Comment by elcreativo** [comment]
Wie würdest du das aktuell über das Template lösen? Die Begrenzung der Standorte in den Einstellungen des Moduls auf z.b. 5 maximal funktioniert bei mir aktuell nicht.

**Comment by Tastaturberuf** [comment, marked as answer]
Ganz oben in `mod_anystores_list`:

``` php
<?php

    $arrStores = [];
    $arrIds    = array_rand($this->stores, 3);

    foreach ( $arrIds as $intId )
    {
        $arrStores[] = $this->stores[$intId];
    }

    $this->stores = $arrStores;

?>
```

Das holt 3 zufällige IDs aus dem Array. Erstellt ein temporäres Array mit den zufälligen Standorten und überschreibt das normale Standort-Array damit der Rest des Templates nicht angepasst werden muss. 

Das LIMIT-Attribut wird bisher nur bei Suchabfragen berücksichtigt, und spielt deswegen keine Rolle.
